cluster_experiments/experiment_analysis.py

```diff
@@ -680,3 +682,226 @@ def analysis_point_estimate(self, df: pd.DataFrame, verbose: bool = False) -> fl
+        treatment (str): The name of the treatment to use as the treated group.
+        cluster_cols (list): A list of columns to use as clusters.
+        hypothesis (str): One of "two-sided", "less", "greater" indicating the hypothesis.
+        time_col (str): The name of the column containing the time data.
+        intervention_date (str): The date when the intervention occurred.
+    Usage:
+
+    ```python
+    from cluster_experiments.experiment_analysis import SyntheticControlAnalysis
+    import pandas as pd
+    import numpy as np
+
+    dates = pd.date_range(start_date, end_date, freq="d")
+
+    users = [f"User {i}" for i in range(N)]
+
+    # Create a combination of each date with each user
+    combinations = list(product(users, dates))
+
+    target_values = np.random.normal(0, 1, size=len(combinations))
+
+    df = pd.DataFrame(combinations, columns=["user", "date"])
+    df["target"] = target_values
+
+
+    SyntheticControlAnalysis(
+        cluster_cols=["user"], time_col="date", transition_date="2022-01-15"
+    ).get_pvalue(df)
+
+    ```
+    """
+
+    def __init__(
+        self,
+        target_col: str = "target",
+        treatment_col: str = "treatment",
+        treatment: str = "B",
+        cluster_cols: Optional[List[str]] = None,
+        hypothesis: str = "two-sided",
+        time_col: str = "date",
+        intervention_date: str = None,
+    ):
+        super().__init__(cluster_cols)
+
+        self.target_col = target_col
+        self.treatment = treatment
+        self.treatment_col = treatment_col
+        self.cluster_cols = cluster_cols
+        self.hypothesis = hypothesis
+        self.time_col = time_col
+        self.intervention_date = intervention_date
+
+    # todo  check is that time_col not in cluster_cols
+
+    def get_pvalue(self, df: pd.DataFrame) -> float:
+        """Returns the p-value of the analysis
+
+        Arguments:
+            df: dataframe containing the data to analyze
+        """
+        df = df.copy()
+        df = self._create_binary_treatment(df)
+        self._data_checks(df=df)
+        return self.analysis_pvalue(df=df)
+
+    def fit_synthetic(self, pre_experiment_df: pd.DataFrame, verbose: bool) -> list:
+        """Returns the weights of each donor"""
+
+        if not any(pre_experiment_df[self.treatment_col] == 1):
+            raise ValueError("No treatment unit found in the data.")
+
+        X = (
+            pre_experiment_df.query(f"{self.treatment_col}==0")
+            .pivot(index=self.cluster_cols, columns=self.time_col)[self.target_col]
+            .T
+        )
+
+        y = (
+            pre_experiment_df.query(f"{self.treatment_col}==1")
+            .pivot(index=self.cluster_cols, columns=self.time_col)[self.target_col]
+            .T.iloc[:, 0]
+        )
+
+        weights = get_w(X, y, verbose)
+
+        return weights
+
+    def fit_predict_synthetic(
+        self,
+        pre_experiment_df: pd.DataFrame,
```

Comment:
any reason why pre_df is first argument? for consistency with the library, I'd add df as first argument